src/Tes/Repository/PostgreSqlCachingRepository.cs

```diff
@@ -97,6 +101,48 @@ protected async Task<IEnumerable<T>> GetItemsAsync(DbSet<T> dbSet, Expression<Fu
             return await _asyncPolicy.ExecuteAsync(ct => query.ToListAsync(ct), cancellationToken);
         }
 
+        protected async Task<IEnumerable<TesTaskDatabaseItem>> GetTesTaskDatabaseItemsByTagAsync(
+            TesDbContext dbContext,
+            Dictionary<string, string> tags,
+            CancellationToken cancellationToken)
+        {
+            ArgumentNullException.ThrowIfNull(dbContext);
+            ArgumentNullException.ThrowIfNull(tags);
+            if (tags.Count == 0) throw new ArgumentOutOfRangeException("Must specify more than one tag");
+
+            var sqlBuilder = new StringBuilder();
+            sqlBuilder.AppendLine("SELECT * FROM TesTasks WHERE ");
+            var tagConditions = tags.Select(kvp => $"\"Json\"->'Tags'->>'{kvp.Key}' = @p_{kvp.Key}");
```

Comment:
To prevent SQL injection, should we do any validation on each `Key`? Or, can we pass in the keys as command parameters?

Reply:
Agreed will need to research this more.  Will depend on what Npgsql supports.  Presumably there is a way to do this. but not sure since it's a JSON query